samcli/lib/cookiecutter/question.py

```diff
@@ -0,0 +1,134 @@
+        can be used in a flow like:
+            supposed we are prompting the user for two resources r1 & r2 such that if the resource already
+            exist then the user provide it otherwise we created it on his behave. the questions may look like:
+                Question(key="r1" text="do you already have resource1?",
+                         _next_question_map={"True": "r2", "False": "r1-creation"}
+                Question(key="r1-creation" text="What do you like to name resource1?"}
+                Question(key="r2" text="do you already have resource2?",
+                         _next_question_map={"True": "r3", "False": "r2-creation"}
+    _default_next_question_key: str
+        The key of the next question that is not based on user's answer, i.e., if there is no matching in
+        the _next_question_map
+    _kind: QuestionKind
+        The kind of the question, it can be one of the values of the QuestionKind enum. This basically directs
+        click on how to prompt the user for the answer; click.prompt, click.confirm, click.choice ...etc
+    """
+
+    def __init__(
+        self,
+        key: str,
+        text: str,
+        options: Optional[List[str]] = None,
+        default: Optional[str] = None,
+        is_required: Optional[bool] = False,
+        next_question_map: Optional[Dict[str, str]] = None,
+        default_next_question_key: Optional[str] = None,
+        kind: Optional[QuestionKind] = None,
+    ):
+        self._key = key
+        self._text = text
+        self._options = options or []
+        self._required = is_required
+        self._default_answer = default
+        # if it is an optional question, set an empty default answer to prevent click from keep asking for an answer
+        if not self._required and self._default_answer is None:
+            self._default_answer = ""
+        self._next_question_map = next_question_map or {}
+        self._default_next_question_key = default_next_question_key
+        self._kind = kind or (QuestionKind.confirm if options else QuestionKind.default)
+
+    @property
+    def key(self):
+        return self._key
+
+    @property
+    def text(self):
+        return self._text
+
+    @property
+    def options(self):
+        return self._options
+
+    @property
+    def default_answer(self):
+        return self._default_answer
+
+    @property
+    def required(self):
+        return self._required
+
+    @property
+    def next_question_map(self):
+        return self._next_question_map
+
+    @property
+    def default_next_question_key(self):
+        return self._default_next_question_key
+
+    @property
+    def kind(self):
+        return self._kind
+
+    def get_choices_indexes(self, base: int = 0) -> List[str]:
+        if self._options:
+            choices = list(range(base, len(self._options) + base))
+            return list(map(str, choices))
+        return list()
+
+    def get_option(self, index) -> Any:
```

Comment:
Should this be `-> str`?

Reply:
tried a lot with no success. mypy keep complaining that `self._options[index]` is of type `Any` and I have no idea why.

Reply:
Change this to this in your `__init__()` should resolve it

```
self._options = options or []
->
self._options: List[str] = options or []
```

Reply:
`options or []` returns `Union[List[str], List[Any]]`, maybe that's why you keep getting that error.

Reply:
> Change this to this in your `__init__()` should resolve it
> 
> ```
> self._options = options or []
> ->
> self._options: List[str] = options or []
> ```

I tried this yesterday, didn't work, let me give it a 2nd try.

Reply:
I think now it can be a `-> str`

Reply:
```suggestion
    def get_option(self, index) -> str:
```